File: extensions/python-runtime/runtime/src/main/java/dev/ujhhgtg/wekit/python/runtime/RuntimeEntrypoint.java
package dev.ujhhgtg.wekit.python.runtime;

import dev.ujhhgtg.wekit.python.api.PythonPluginHost;
import dev.ujhhgtg.wekit.python.api.PythonPluginRequest;
import dev.ujhhgtg.wekit.python.api.PythonRuntimeApi;
import dev.ujhhgtg.wekit.python.api.PythonRuntimeBackend;
import dev.ujhhgtg.wekit.python.api.PythonRuntimeConfig;

/**
 * Loader-neutral placeholder for the runtime pack bootstrap.
 *
 * Native loading and the backend are intentionally left for the runtime
 * implementation task. Keeping this class free of implementation references is
 * required so reflection can verify the contract before any native load.
 */
public final class RuntimeEntrypoint {
    private RuntimeEntrypoint() { }

    public static PythonRuntimeBackend bootstrap(
            int apiVersion,
            PythonRuntimeConfig config,
            PythonPluginHost host) {
        if (apiVersion != PythonRuntimeApi.API_VERSION) {
            throw new IllegalArgumentException(
                    "Unsupported Python runtime API version: " + apiVersion);
        }
        if (config == null || host == null) {
            throw new IllegalArgumentException("Runtime config and host are required");
        }
        return new UnavailableBackend();
    }

    private static final class UnavailableBackend implements PythonRuntimeBackend {
        private UnsupportedOperationException unavailable() {
            return new UnsupportedOperationException(
                    "Python runtime backend is not implemented in Task 1; native bootstrap was not attempted");
        }

        @Override public void start(PythonRuntimeConfig config) { throw unavailable(); }
        @Override public void activatePlugin(PythonPluginRequest request, PythonPluginHost host) { throw unavailable(); }
        @Override public void deactivatePlugin(String pluginId) { throw unavailable(); }
        @Override public void reloadPlugin(PythonPluginRequest request, PythonPluginHost host) { throw unavailable(); }
    }
}

File: extensions/python-runtime/runtime/src/main/kotlin/dev/ujhhgtg/wekit/python/runtime/RuntimeEntrypoint.kt
package dev.ujhhgtg.wekit.python.runtime

import dev.ujhhgtg.wekit.python.api.PythonPluginHost
import dev.ujhhgtg.wekit.python.api.PythonPluginRequest
import dev.ujhhgtg.wekit.python.api.PythonRuntimeApi
import dev.ujhhgtg.wekit.python.api.PythonRuntimeBackend
import dev.ujhhgtg.wekit.python.api.PythonRuntimeConfig

/** Loader-neutral placeholder; native/backend implementation belongs to later tasks. */
object RuntimeEntrypoint {
    @JvmStatic
    fun bootstrap(apiVersion: Int, config: PythonRuntimeConfig, host: PythonPluginHost): PythonRuntimeBackend {
        require(apiVersion == PythonRuntimeApi.API_VERSION) { "Unsupported Python runtime API version: $apiVersion" }
        requireNotNull(config) { "Runtime config is required" }
        requireNotNull(host) { "Runtime host is required" }
        return UnavailableBackend()
    }

    private class UnavailableBackend : PythonRuntimeBackend {
        private fun unavailable() = UnsupportedOperationException(
            "Python runtime backend is not implemented in Task 1; native bootstrap was not attempted",
        )
        override fun start(config: PythonRuntimeConfig): Unit = throw unavailable()
        override fun activatePlugin(request: PythonPluginRequest, host: PythonPluginHost): Unit = throw unavailable()
        override fun deactivatePlugin(pluginId: String): Unit = throw unavailable()
        override fun reloadPlugin(request: PythonPluginRequest, host: PythonPluginHost): Unit = throw unavailable()
    }
}
